### PR TITLE
[BEAM-1258] BigQueryServicesImpl.getTable() returns null when tables not found.

### PR DESCRIPTION
**Pull request by peihe**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by peihe** [comment]
R: @dhalperi

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java [hunk 1]
```diff
@@ -398,14 +398,30 @@ public Table getTable(String projectId, String datasetId, String tableId)
...
-          Sleeper.DEFAULT,
+      return getTable(
+          new TableReference().setProjectId(projectId).setDatasetId(datasetId).setTableId(tableId),
           backoff,
-          DONT_RETRY_NOT_FOUND);
+          Sleeper.DEFAULT);
+    }
+
+    @VisibleForTesting
+    Table getTable(TableReference ref, BackOff backoff, Sleeper sleeper)
```
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java]: mark as `@Nullable`

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java [hunk 2]
```diff
@@ -398,14 +398,30 @@ public Table getTable(String projectId, String datasetId, String tableId)
       BackOff backoff =
           FluentBackoff.DEFAULT
               .withMaxRetries(MAX_RPC_RETRIES).withInitialBackoff(INITIAL_RPC_BACKOFF).backoff();
-      return executeWithRetries(
-          client.tables().get(projectId, datasetId, tableId),
-          String.format(
-              "Unable to get table: %s, aborting after %d retries.",
-              tableId, MAX_RPC_RETRIES),
-          Sleeper.DEFAULT,
+      return getTable(
```
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java]: mark function `@Nullable` and clarify javadoc?

Comment (peihe) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java]: done
And, the {@inheritDoc} explained the return value null.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java (hunk 1 above)
Comment (peihe) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java]: done

**Comment by dhalperi** [comment]
LGTM, thanks!

**Pull request closed without merging**